core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -565,6 +565,9 @@ void ensureRootConverters() {
       }
     }
     for (RelSubset subset : root.set.subsets) {
+      if (subset == root) {
```

Comment:
I'm not sure whether the logic of these lines of code is correct, could you add a unit test for this logic?